src/cli.cc

```diff
@@ -96,6 +109,30 @@ ExitCode CompleteInstall(AkliteClient& client) {
       exit_code = ExitCode::InstallNeedsRebootForBootFw;
       break;
     }
+    case InstallResult::Status::Failed: {
+      // do rollback
+      // check rollback type, bootloader (boot on the previous version) or App driven
+      const auto current{client.GetCurrent()};  // returns Target a device is booted on
```

Comment:
No. The `GetRollbackTarget` works only for the "app driven" rollback.  If this is "ostree" rollback (the bootloader did rollback and a device booted on the previous target) then we need just sync "current" since the "current" is always bound to the ostree version a device is booted  on.

What I've been considering, while implementing it, is to hide the difference between the "app driver" and "ostree" rollback behind `GetRollbackTarget`. Specifically, its implementation could return the rollback target in the first case and the current one in the second case.